src/modules/Game.class.js

```diff
@@ -20,25 +20,113 @@ class Game {
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.initialState = initialState;
+    this.board = initialState.map((row) => [...row]);
+    this.score = 0;
+    this.isGameStarted = false;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  moveLeft() {
+    const previousBoard = this.getState();
+
+    this.board.forEach((row, index) => {
+      const nonZeroElements = row.filter((value) => value !== 0);
+      const combinedRow = this.joinValues(nonZeroElements, row.length);
+
+      this.board[index] = combinedRow;
+    });
+
+    if (this.isBoardChanged(previousBoard, this.board)) {
+      const randomCell = this.getRandomCell(1);
+
+      this.addNewCell(randomCell);
+    }
+  }
+
+  moveRight() {
+    const previousBoard = this.getState();
+
+    this.board.forEach((row, index) => {
+      const reversedRow = row.slice().reverse();
+
+      const nonZeroElements = reversedRow.filter((value) => value !== 0);
+      const combinedRow = this.joinValues(nonZeroElements, row.length);
+
+      this.board[index] = combinedRow.reverse();
+    });
+
+    if (this.isBoardChanged(previousBoard, this.board)) {
+      const randomCell = this.getRandomCell(1);
+
+      this.addNewCell(randomCell);
+    }
+  }
+
+  moveUp() {
+    const previousBoard = this.getState();
+
+    const transposedBoard = this.transpose(this.board);
+
+    transposedBoard.forEach((row, index) => {
+      const nonZeroElements = row.filter((value) => value !== 0);
+      const combinedRow = this.joinValues(nonZeroElements, row.length);
+
+      transposedBoard[index] = combinedRow;
+    });
+
+    this.board = this.transpose(transposedBoard);
+
+    if (this.isBoardChanged(previousBoard, this.board)) {
+      const randomCell = this.getRandomCell(1);
+
+      this.addNewCell(randomCell);
+    }
```

Comment:
This logic is similar in every method, so it would be better to move it to a helper function and use it everywhere.